internal/app/machined/pkg/controllers/network/node_address.go

```diff
@@ -127,9 +127,17 @@ func (ctrl *NodeAddressController) Run(ctx context.Context, r controller.Runtime
 				continue
 			}
 
+			// in IPv6 only environments we want to prefer longer prefixes, in other environments we keep old behavior
```

Comment:
I wonder if the proper fix is to re-sort the `addresses` list in the first place. Today the order comes from the alphabetical order of address resource IDs which look like `<link>/address/bits` so that the IPv6 addresses for the same link are sorted with longest prefix first?

Reply:
This way it will not only affect `defaultAddress`, but all other list of addresses which are used to pick up etcd, and other listening addresses

Reply:
I am not overly familiar with the Talos codebase, is the `defaultAddress` not used by etcd as well?

Sorting the addresses was my initial thought, but I shied away from that approach as I was uncertain how to handle Dual-Stack environments properly. As a first step I could however change the sorting for IPv6 Single-Stack environments.

Another third option, but larger change might be to introduce CEL or [expr](https://github.com/expr-lang/expr) to allow users to customize filtering and IP selection.

Reply:
Let me think about this more, but my gut feeling is that sorting by string value (as it is today) is not quite good

Reply:
I will have another look this evening and/or tomorrow as I also believe sorting might be the better approach. Grepping the codebase once more I am not certain I even understand what the purpose of defaultAddress is as it is only used in the hostname and etcfile controller (today, might be a relic of the past).